pkg/machine/shim/host.go

```diff
@@ -636,38 +637,110 @@ func confirmationMessage(files []string) {
 	}
 }
 
-func Reset(dirs *machineDefine.MachineDirs, mp vmconfigs.VMProvider, mcs map[string]*vmconfigs.MachineConfig) error {
-	var resetErrors *multierror.Error
-	for _, mc := range mcs {
-		err := Stop(mc, mp, dirs, true)
+func Reset(mps []vmconfigs.VMProvider, opts machine.ResetOptions) error {
+	toRemove := []*providerMachines{}
+
+	for _, provider := range mps {
+		pm, err := getMachinesFromProvider(provider)
 		if err != nil {
-			resetErrors = multierror.Append(resetErrors, err)
+			return err
 		}
-		_, genericRm, err := mc.Remove(false, false)
+		toRemove = append(toRemove, pm)
+	}
+
+	if !opts.Force {
+		resetConfirmationMessage(toRemove)
```

Comment:
Why did we move all the printing code here? Style wise I think we should avoid as much as possible printing in the "backend" code. I know machine already ignores this elsewhere but I like it much more to have the prompts/prints under cmd/podman?

Reply:
There was a precedent in Remove() so I did this mostly to avoid doing the work of looking for the machines twice - if this is done in cmd/podman, we'd have to loop through the providers and machines there to get the names, and again in Reset. Though this shouldn't be too expensive, so if you're okay with  the code dup, I can do that? The other option to avoid this would be to have Reset take the providerMachines struct as an arg, which I was initially reluctant to do since it's not a standard arg like the other functions in shim/host.go.

Let me know if either of these sounds better, or to leave it as is.

Reply:
Moved to cmd/podman!